app/views/contexts/section_summary_context.py

```diff
@@ -10,8 +10,8 @@
 
 
 class SectionSummaryContext(Context):
-    def __call__(self, current_location):
-        summary = self._build_summary(current_location)
+    def __call__(self, current_location, return_to="summary"):
```

Comment:
This is a pain and I am not particularly happy with it. As questionnaire summary (i.e final summary) produces multiple section summaries to display content, I have had to pass "return_to" down through the layers to eventually the block (summary) which creates the link.  I couldn't find an alternative way.

Reply:
I don't think there is another way without further refactoring.